clients/include/testing_reduction_batched.hpp

```diff
@@ -0,0 +1,201 @@
+/* ************************************************************************
+ * Copyright 2018-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_cblas.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+template <typename T, typename R>
+using rocblas_reduction_batched_t = rocblas_status (*)(rocblas_handle  handle,
+                                                       rocblas_int     n,
+                                                       const T* const* x,
+                                                       rocblas_int     incx,
+                                                       rocblas_int     batch_count,
+                                                       R*              result);
+
+template <typename T, typename R>
+void template_testing_reduction_batched_bad_arg(const Arguments&                  arg,
+                                                rocblas_reduction_batched_t<T, R> func)
+{
+    rocblas_int N = 100, incx = 1, batch_count = 5;
+
+    rocblas_local_handle handle;
+
+    //
+    // allocate memory on device
+    //
+    device_batch_vector<T> dx(N, incx, batch_count);
+    CHECK_HIP_ERROR(dx.memcheck());
+
+    R h_rocblas_result;
+
+    EXPECT_ROCBLAS_STATUS(func(handle, N, nullptr, incx, batch_count, &h_rocblas_result),
+                          rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(func(handle, N, dx, incx, batch_count, nullptr),
+                          rocblas_status_invalid_pointer);
+
+    EXPECT_ROCBLAS_STATUS(func(nullptr, N, dx, incx, batch_count, &h_rocblas_result),
+                          rocblas_status_invalid_handle);
+}
+
+template <typename T, typename R>
+void template_testing_reduction_batched(const Arguments&                  arg,
+                                        rocblas_reduction_batched_t<T, R> func,
+                                        void (*CBLAS_FUNC)(rocblas_int, const T*, rocblas_int, R*))
+{
+    rocblas_int          N = arg.N, incx = arg.incx, batch_count = arg.batch_count;
+    rocblas_stride       stride_x = arg.stride_x;
+    double               rocblas_error_1, rocblas_error_2;
+    rocblas_local_handle handle;
+
+    if(batch_count < 0)
+    {
+        device_batch_vector<T> dx(10, 1, 10);
+        CHECK_HIP_ERROR(dx.memcheck());
+        device_vector<R> dr(10);
+        CHECK_HIP_ERROR(dr.memcheck());
+
+        CHECK_ROCBLAS_ERROR(rocblas_set_pointer_mode(handle, rocblas_pointer_mode_host));
+        EXPECT_ROCBLAS_STATUS((func(handle, N, dx.ptr_on_device(), incx, batch_count, dr)),
```

Comment:
Parentheses are not needed around `(func(...))`.

Extra parentheses are only required when the outer expression in a macro argument contains commas not surrounded by parentheses, braces or backets, such as `func<T**,**U>(arg1, arg2)`. That needs to be parenthesized at least as `(func<T,U>)(arg1, arg2)` but you can use `(func<T,U>(arg1, arg2))` if you are more comfortable with it. When only one template argument is needed, such as `func<T>`, then no parentheses are necessary because there is no comma unsurrounded by parentheses, brackets or braces in the top-level expression.

Reply:
mistake